src/doc_builder/build_embeddings.py

```diff
@@ -0,0 +1,439 @@
+                    page_info=page_info,
+                    version_tag_suffix=version_tag_suffix,
+                )
+                markdown_chunks = create_markdown_chunks(
+                    content,
+                    page_info=page_info,
+                )
+
+                # Make sure we clean up for next page.
+                del page_info["page"]
+
+                page_chunks = markdown_chunks + autodoc_chunks
+
+                if is_python_module:
+                    page_chunks = [
+                        chunk._replace(text=resolve_links_in_text(chunk.text, package, anchor_mapping, page_info))
+                        for chunk in page_chunks
+                    ]
+
+                chunks.extend(page_chunks)
+
+        except Exception as e:
+            raise type(e)(f"There was an error when converting {file} to chunks to embed.\n" + e.args[0]) from e
+
+        if new_anchors is not None:
+            page_name = str(file.with_suffix("").relative_to(doc_folder))
+            for anchor in new_anchors:
+                if isinstance(anchor, tuple):
+                    anchor_mapping.update(
+                        {a: f"{page_name}#{anchor[0]}" for a in anchor[1:] if a not in anchor_mapping}
+                    )
+                    anchor = anchor[0]
+                anchor_mapping[anchor] = page_name
+
+        if errors is not None:
+            all_errors.extend(errors)
+
+    if len(all_errors) > 0:
+        raise ValueError(
+            "The deployment of the documentation will fail because of the following errors:\n" + "\n".join(all_errors)
+        )
+
+    return chunks
+
+
+def call_embed_api(chunks: List[Chunk]) -> List[Embedding]:
+    """
+    Using https://huggingface.co/inference-endpoints with a text embedding model
+    """
+    batch_size = 32
+    chunks_len = len(chunks)
+    embeddings = []
+
+    def query(payload):
+        response = requests.post(API_URL, headers=headers, json=payload)
+        return response.json()
+
+    # warm up API call
+    output_warmup = query({"inputs": "Hello World!"})
+    if isinstance(output_warmup, dict) and "error" in output_warmup:
+        if output_warmup["error"] == "503 Service Unavailable":
+            print("Waking up Embedding Inference Endpoints. Retrying in 5 minutes")
+            time.sleep(300)  # 5 minutes
+        else:
+            raise Exception("Embedding Inference Endpoints error", output_warmup["error"])
+
+    for i in tqdm(range(0, chunks_len, batch_size), desc="Calling Embedding Inference Endpoints"):
+        batch_chunk = chunks[i : i + batch_size]
+        batch_chunk_texts = [chunk.text for chunk in batch_chunk]
+
+        API_URL = os.environ["HF_EMBED_URL"]
+        headers = {
+            "Accept": "application/json",
+            "Authorization": f"Bearer {os.environ['HF_EMBED_API_KEY']}",
+            "Content-Type": "application/json",
+        }
+
+        batch_embeddings = query({"inputs": batch_chunk_texts})
```

Comment:
great suggestion! Yes, I'm using huggingface_hub https://github.com/huggingface/doc-builder/pull/506/commits/58caef8fcecf2d50cfa8b954635d5c811ce78502 now. Otherwise, I'd need re-create all error handlings etc that are already done in huggingface_hub